project/lolainta/README.md

```diff
@@ -0,0 +1,110 @@
+
+## System Architecture
+
+The data are provided with json format. Thus I will chose to use python to read the data first, since disk I/O is the main reason for slowing down this stage. And using python to parse json format file is more convenient, of course.
+
+Then, I will manipulate the data with cpp, include indexing them or process any inquiries when called with pyhton API.
+
+As mentioned above, I will wrap it with Python API in order to support users using python API before.
+
+## API Description
+
+```python
+import nuscenes
+
+if __name__ == "__main__":
+    nusc = nuscenes.Nuscene(dataroot="/path/to/the/data", version="mini")
+    # get scene data by token in scene table
+    scene: nuscenes.Scene = nusc.scene("token")
+    # same as others
+    anno: nuscenes.Annotation = nusc.annotation("token")
+    sample: nuscenes.Sample = nusc.sample("token")
+    # ...
+
+    # get scene data by index in scene table
+    scene: nuscenes.Scene = nusc.scenes[0]
+    # same as others
+    anno: nuscenes.Annotation = nusc.annotations[0]
+    sample: nuscenes.Sample = nusc.samples[0]
+    # ...
+
+    # get sample associated with specific scene
+    # This sample object will contain informations in sample_data, ego_pose, calibrated_sensor, and timestamp
+    sample: nuscenes.Sample = scene.samples
+    # get vehicle log associated with specific scene
+    log: nuscenes.Log = scene.log
+    # get annotation associated with specific sample
+    # This annotation object will contain informations in category, attribute, visibility, instance, and sample
+    anno: nuscenes.Annotation = sample.annotations
+    # get instance associated with specific annotation
+    # This instance object will contain informations in category, attribute, visibility, and sample_annotation
+    instance: nuscenes.Instance = anno.instance
+
+```
+
+
+
+## Engineer Infrastructure
+
+- Containerlized Building Environment
+  - [docker](https://www.docker.com)
+  - [GNU make](https://www.gnu.org/software/make/manual/make.html)
+- Testing Framework
+  - [pytest](https://docs.pytest.org/en/7.4.x/index.html) for python
+  - [GoogleTest](https://github.com/google/googletest) for C++ (hopefully)
+- Documentation
+  - [ReadTheDocs](https://about.readthedocs.com/?ref=readthedocs.org) (hopefully)
+  - [pydoc](https://docs.python.org/3/library/pydoc.html)
+  - Markdown
+- Version Control System
+  - git
+- Continuous Integration
+  - [Github Actions](https://github.com/features/actions)
+
+## Schedule 
+
+The development of the C++ SDK for the nuScenes dataset will follow a well-structured timeline, ensuring that key functions are implemented efficiently while allowing for flexibility to accommodate evolving requirements.
+
+- **Week 1 (10/31): Performance Analysis**
+  - Conduct an in-depth analysis of the performance of the existing Python SDK to identify specific areas for optimization.
```

Comment:
How do you want to analyze the performance?  For example, in what format do you want to make reports?

Reply:
Maybe by simply calling a function many times and measure its execution time.

Reply:
The format of your profiling report more or less determines how you analyze the performance.  Please think about it.